pkg/occlient/occlient_test.go

```diff
@@ -2080,3 +2121,86 @@ func Test_getContainerPortsFromStrings(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateService(t *testing.T) {
+	tests := []struct {
+		name             string
+		commonObjectMeta metav1.ObjectMeta
+		containerPorts   []corev1.ContainerPort
+		wantErr          bool
+	}{
+		{
+			name: "Test case: with valid commonObjectName and containerPorts",
+			commonObjectMeta: metav1.ObjectMeta{
+				Name: "nodejs",
+				Labels: map[string]string{
+					"app": "apptmp",
+					"app.kubernetes.io/component-name": "ruby",
+					"app.kubernetes.io/component-type": "ruby",
+					"app.kubernetes.io/name":           "apptmp",
+				},
+				Annotations: map[string]string{
+					"app.kubernetes.io/url":                   "https://github.com/openshift/ruby",
+					"app.kubernetes.io/component-source-type": "git",
+				},
+			},
+			containerPorts: []corev1.ContainerPort{
+				{
+					Name:          "8080-tcp",
+					ContainerPort: 8080,
+					Protocol:      corev1.ProtocolTCP,
+				},
+				{
+					Name:          "9100-udp",
+					ContainerPort: 9100,
+					Protocol:      corev1.ProtocolUDP,
+				},
+			},
+		},
+	}
```

Comment:
We'll need another case with invalid values.

Reply:
sadly we can't, as the fake client doesn't do any validations.

Reply:
@ashetty1 I can't think of anything, can you give an example for a invalid case?

Reply:
@kadel @mik-dass empty values validation?

Reply:
@ashetty1 I tried but it didn't work for me

Reply:
Nevertheless valid case, no?